### ISSUE TITLE
Component hierarchy and tag naming

### ISSUE THREAD

**Issue by aalaap** [spatie/laravel-blade-x#55]
I have some components in a structure under views:

    frontend/components/item.blade.php
    frontend/components/widgets/search.blade.php
    frontend/components/modals/save.blade.php

If I register all these, I am able to use each component as a top-level tag:

    <x-item id="25">...</x-item>
    <x-search section="Events"></x-search>
    <x-save></x-save>  

In the template files, it isn't clear upfront if I'm using a widget, a modal or a regular partial. Is there any way to overcome this problem without naming the component something like `searchWidget.blade.php`? I am wondering if it is a good idea to indicate a "root" components path and use folder names below that path as prefixes, so we get...

    <x-item id="25">...</x-item>
    <x-widgets-search section="Events"></x-widgets-search>
    <x-modals-save></x-widgets-save>

**Comment by freekmurze** [comment]
If you need to customize the tag name you can do so with the `tag` method.

```php
BladeX::component('components. search')->tag('search-widget');
```

We're not going to add a `root` path in the package, because I suspect the majority of users won't need this and we like to keep the package simple.

**Comment by aalaap** [comment]
Thanks for this. I did look at the `tag` method in the documentation, but I realised it still needs to be specified for each component separately, so if I use the `component` method to register all components in a folder, that's not going to work.

Another suggestion is allowing to set a prefix within the `component()` method, similar to route grouping prefixes. Something like:

    BladeX::component('components.widgets.*', ['prefix' => 'widget']);

This could make every component under the `widgets` folder have the prefix, for eg. `widget-search`, without having to specify it one by one. The `prefix` method can still be used to set a global prefix, for eg. `my`, which would then make the complete component tag `my-widget-search`.
